src/backend/mod.rs

```diff
@@ -1,25 +1,45 @@
+//! Contains the different backends to get the screenshot from.
+
 pub mod wayland;
-pub mod xorg;
+pub mod x11;
+
+/// A general backend error enum.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// Represents that an error occured while trying to get a screenshot on X11.
+    #[error(transparent)]
+    X11(#[from] x11::Error),
+}
 
 pub type Pixel = u16;
 
-pub enum OutputIdentifier {
-    X11(u32),
-    Wayland {
-        id: u32,
-        name: String,
-        description: String,
-    },
+#[derive(Debug)]
+pub enum AdditionalInformation {
+    X11 { name: u32 },
+    Wayland { name: String, description: String },
 }
 
+#[derive(Debug)]
 pub struct OutputInfo {
-    pub identifier: OutputIdentifier,
     pub width: Pixel,
     pub height: Pixel,
     pub x: i16,
     pub y: i16,
+
+    pub id: u32,
+    pub info: AdditionalInformation,
```

Comment:
OutputInfo has a field info? Feels a bit off. Why not monitor/monitorAttributes or so?

Reply:
done!